src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -972,6 +972,48 @@ public async Task Alpn_H3_Success()
             await connection.DisposeAsync();
         }
 
+        [Fact]
+        public async Task H3_AltUsedHeader()
+        {
+            var options = new Http3Options() { Alpn = SslApplicationProtocol.Http3.ToString() };
+            using Http3LoopbackServer server = CreateHttp3LoopbackServer(options);
+
+            Http3LoopbackConnection connection = null;
+            HttpRequestData requestData = null;
+            Task serverTask = Task.Run(async () =>
+            {
+                connection = (Http3LoopbackConnection)await server.EstablishGenericConnectionAsync();
+                requestData = await connection.ReadRequestDataAsync(readBody: false);
+                Console.WriteLine(requestData);
+                await connection.SendResponseAsync();
+
+
+            });
+
+            using HttpClient client = CreateHttpClient();
+            using HttpRequestMessage request = new()
+            {
+                Method = HttpMethod.Get,
+                RequestUri = server.Address,
+                Version = HttpVersion30,
+                VersionPolicy = HttpVersionPolicy.RequestVersionExact
+            };
+
+           HttpResponseMessage response = await client.SendAsync(request).WaitAsync(TimeSpan.FromSeconds(10));
+            
+            
+            response.EnsureSuccessStatusCode();
+            Assert.Equal(HttpVersion.Version30, response.Version);
+
+            await serverTask;
+            Assert.NotNull(connection);
+            Assert.DoesNotContain("alt-used", requestData.ToString());
```

Comment:
You should be able to get the headers via `requestData.Headers` and from there you can find - rather not find - a match by `HttpHeaderData.Name`. Look around `HttpRequestData` data structure.